canvas_site_deletion/views.py

```diff
@@ -161,3 +174,33 @@ def delete(request, pk):
     messages.success(request, f"Successfully deleted Canvas course {canvas_course_id} and cleaned up course_instance {pk}")
 
     return render(request, 'canvas_site_deletion/index.html')
+
+
+def get_account_hierarchy(course_id) -> str:
+    """
+    Get the list of accounts associated with a given course ID
+    """
+    account_ids = []
+
+    # use course id to get account id
+    canvas_course = canvas_get_course(SDK_CONTEXT, course_id).json()
+    canvas_course_account_id = canvas_course["account_id"] 
+    account_ids.append(canvas_course_account_id)
+
+    # traverse account hierarchy in both directions
+    # first downwards to find sub-accounts
+    sub_accounts = get_sub_accounts_of_account(SDK_CONTEXT, canvas_course_account_id).json()
+    for sub_account in sub_accounts:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)